pkg/controller/activegate/activegate_controller.go

```diff
@@ -156,6 +168,23 @@ func (r *ReconcileActiveGate) Reconcile(request reconcile.Request) (reconcile.Re
 	return builder.ReconcileAfterFiveMinutes(), nil
 }
 
+func (r *ReconcileActiveGate) reconcilePullSecret(instance *dynatracev1alpha1.ActiveGate, log logr.Logger, dtc dtclient.Client) error {
+	var tkns corev1.Secret
+	if err := r.client.Get(context.TODO(), client.ObjectKey{Name: parser.GetTokensName(instance), Namespace: instance.GetNamespace()}, &tkns); err != nil {
+		return fmt.Errorf("failed to query tokens: %w", err)
+	}
+	pullSecretData, err := builder.GeneratePullSecretData(instance, dtc, &tkns)
+	if err != nil {
+		return fmt.Errorf("failed to generate pull secret data: %w", err)
+	}
+	err = factory.CreateOrUpdateSecretIfNotExists(r.client, r.client, instance.GetName()+"-pull-secret", instance.GetNamespace(), pullSecretData, corev1.SecretTypeDockerConfigJson, log)
+	if err != nil {
+		return fmt.Errorf("failed to create or update secret: %w", err)
+	}
+
+	return nil
+}
+
```

Comment:
imo should be moved into it's own file, i.e. `activegate/pull_secret.go` to not bloat `activegate_controller.go` file